packages/qwik/src/optimizer/core/src/transform.rs

```diff
@@ -1389,7 +1386,16 @@ impl<'a> QwikTransform<'a> {
 											}),
 										}),
 									));
-									event_handlers.push(converted_prop);
+									if !is_const {
+										static_listeners = false;
+										var_props.push(converted_prop.clone().fold_with(self));
+									} else if spread_props_count > 0 {
+										maybe_const_props
+											.push(converted_prop.clone().fold_with(self));
+									} else {
+										const_props.push(converted_prop.clone().fold_with(self));
```

Comment:
you can just always push to maybe_const_props, no need to check for the spread props again.